packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.util.ts

```diff
@@ -538,4 +546,21 @@ describe('internal-plugin-metrics', () => {
       });
     });
   });
+
+  describe('generateClientErrorCodeForIceFailure', () => {
+    [
+     { signalingState: 'have-local-offer', iceConnectionState: 'connected', turnServerUsed: true, errorCode: MISSING_ROAP_ANSWER_CLIENT_CODE},
+     { signalingState: 'stable', iceConnectionState: 'connected', turnServerUsed: true, errorCode: DTLS_HANDSHAKE_FAILED_CLIENT_CODE},
+     { signalingState: 'stable', iceConnectionState: 'failed', turnServerUsed: true, errorCode: ICE_FAILED_WITH_TURN_TLS_CLIENT_CODE},
+     { signalingState: 'stable', iceConnectionState: 'failed', turnServerUsed: false, errorCode: ICE_FAILED_WITHOUT_TURN_TLS_CLIENT_CODE},
+     { signalingState: 'failed', iceConnectionState: 'connected', turnServerUsed: false, errorCode: ICE_FAILURE_CLIENT_CODE},
```

Comment:
this case is not realistic, I don't think you can have iceConnectionState "connected" with signalingState "failed"

Reply:
addressed